source/include/sigv4.h

```diff
@@ -110,14 +110,23 @@
  */
 #define SIGV4_HTTP_HEADERS_ARE_CANONICAL_FLAG    0x4U
 
+/**
+ * @ingroup sigv4_canonical_flags
+ * @brief Set this flag to indicate that the HTTP request payload is
+ * already hashed.
+ *
+ * This flag is valid only for #SigV4HttpParameters_t.flags.
+ */
+#define SIGV4_HTTP_PAYLOAD_IS_HASH               0x8U
+
 /**
  * @ingroup sigv4_canonical_flags
  * @brief Set this flag to indicate that the HTTP request path, query, and
- * headers are all already canonicalized.
+ * headers are all already canonicalized and request payload is already hashed.
  *
  * This flag is valid only for #SigV4HttpParameters_t.flags.
  */
-#define SIGV4_HTTP_ALL_ARE_CANONICAL_FLAG        0x7U
+#define SIGV4_HTTP_ALL_ARE_CANONICAL_FLAG        0xFU
```

Comment:
Does canonicity have anything to do with being a hash? This looks like it should be staying `0x7U` and the hash flag staying separate